samples/rtl/axpy_double_pump.py

```diff
@@ -71,412 +71,361 @@
 # add symbol
 N = dace.symbol('N')
 
-# add sdfg
-sdfg = dace.SDFG('axpy_double_pump')
-
-# add state
-state = sdfg.add_state('device_state')
-
-# add parametr
-veclen = 2
-sdfg.add_constant('VECLEN', veclen)
-sdfg.add_constant('DATA_WIDTH', 64)
-sdfg.add_constant('WORD_WIDTH', 32)
-sdfg.add_constant('RATIO', 2)
-
-# add arrays
-sdfg.add_scalar('a', dtype=dace.float32, storage=dace.StorageType.FPGA_Global)
-sdfg.add_array('x', [N // veclen], dtype=dace.vector(dace.float32, veclen), storage=dace.StorageType.CPU_Heap)
-sdfg.add_array('y', [N // veclen], dtype=dace.vector(dace.float32, veclen), storage=dace.StorageType.CPU_Heap)
-sdfg.add_array('result', [N // veclen], dtype=dace.vector(dace.float32, veclen), storage=dace.StorageType.CPU_Heap)
-sdfg.add_array('fpga_x', [N // veclen],
-               dtype=dace.vector(dace.float32, veclen),
-               transient=True,
-               storage=dace.StorageType.FPGA_Global)
-sdfg.add_array('fpga_y', [N // veclen],
-               dtype=dace.vector(dace.float32, veclen),
-               transient=True,
-               storage=dace.StorageType.FPGA_Global)
-sdfg.add_array('fpga_result', [N // veclen],
-               dtype=dace.vector(dace.float32, veclen),
-               transient=True,
-               storage=dace.StorageType.FPGA_Global)
-
-# add streams
-sdfg.add_stream('x_stream',
-                buffer_size=32,
+def make_sdfg():
+    # add sdfg
+    sdfg = dace.SDFG('axpy_double_pump')
+
+    # add state
+    state = sdfg.add_state('device_state')
+
+    # add parametr
+    veclen = 2
```

Comment:
I would have this as an argument and explain what is its meaning here (inward/outward story)

Reply:
This has been implemented.